roachtest: psycopg failed

[(roachtest).psycopg failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2522960&tab=buildLog) on [master@de91557c0634ab3797356d445ba39e37d45d8205](https://github.com/cockroachdb/cockroach/commits/de91557c0634ab3797356d445ba39e37d45d8205):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/psycopg/run_1
	orm_helpers.go:228,orm_helpers.go:154,psycopg.go:122,psycopg.go:135,test_runner.go:760: 
		Tests run on Cockroach v21.1.0-alpha.1-400-gde91557c06
		Tests run against psycopg 2_8_6
		452 Total Tests Run
		450 tests passed
		2 tests failed
		320 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_module.ExceptionsTestCase.test_9_6_diagnostics (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList21_1) is available in the artifacts' psycopg log

```

<details><summary>More</summary><p>


Artifacts: [/psycopg](https://teamcity.cockroachdb.com/viewLog.html?buildId=2522960&tab=artifacts#/psycopg)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Apsycopg.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).psycopg failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2525484&tab=buildLog) on [master@eda9189cecbbc279f1857f6e6b992bdfd363397e](https://github.com/cockroachdb/cockroach/commits/eda9189cecbbc279f1857f6e6b992bdfd363397e):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/psycopg/run_1
	orm_helpers.go:228,orm_helpers.go:154,psycopg.go:122,psycopg.go:135,test_runner.go:760: 
		Tests run on Cockroach v21.1.0-alpha.1-429-geda9189cec
		Tests run against psycopg 2_8_6
		452 Total Tests Run
		450 tests passed
		2 tests failed
		320 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_module.ExceptionsTestCase.test_9_6_diagnostics (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList21_1) is available in the artifacts' psycopg log

```

<details><summary>More</summary><p>


Artifacts: [/psycopg](https://teamcity.cockroachdb.com/viewLog.html?buildId=2525484&tab=artifacts#/psycopg)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Apsycopg.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).psycopg failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2528188&tab=buildLog) on [master@46919380225dba7122130c338744b561d7eb6c56](https://github.com/cockroachdb/cockroach/commits/46919380225dba7122130c338744b561d7eb6c56):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/psycopg/run_1
	orm_helpers.go:228,orm_helpers.go:154,psycopg.go:122,psycopg.go:135,test_runner.go:760: 
		Tests run on Cockroach v21.1.0-alpha.1-455-g4691938022
		Tests run against psycopg 2_8_6
		452 Total Tests Run
		450 tests passed
		2 tests failed
		320 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_module.ExceptionsTestCase.test_9_6_diagnostics (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList21_1) is available in the artifacts' psycopg log

```

<details><summary>More</summary><p>


Artifacts: [/psycopg](https://teamcity.cockroachdb.com/viewLog.html?buildId=2528188&tab=artifacts#/psycopg)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Apsycopg.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cmd/roachtest/psycopg_blocklist.go
+++ b/pkg/cmd/roachtest/psycopg_blocklist.go
@@ -29,7 +29,8 @@ var psycopgBlocklists = blocklistsForVersion{
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
 var psycopgBlockList21_1 = blocklist{
-	"tests.test_async_keyword.CancelTests.test_async_cancel": "41335",
+	"tests.test_async_keyword.CancelTests.test_async_cancel":    "41335",
+	"tests.test_module.ExceptionsTestCase.test_9_6_diagnostics": "58035",
 }
 
 var psycopgBlockList20_2 = blocklist{

